lib/render/AnnotationRenderer.cpp

```diff
@@ -278,40 +279,264 @@ void AnnotationRenderer::applyTransform(Transform *t) {
 	_glManager->matrixManager->Translate(translate[0], translate[1], translate[2]);
 }
 
+void AnnotationRenderer::_makeTransformMatrices(
+    const Transform* transform,
+    glm::mat4 &scalingMatrix,
+    glm::mat4 &rotationMatrix,
+    glm::mat4 &translationMatrix,
+    glm::mat4 &translateOriginMatrix
+) const {
+	vector<double>  scales       = transform->GetScales();
+	vector<double>  origins      = transform->GetOrigin();
+	vector <double> translations = transform->GetTranslations();
+	vector <double> rotations    = transform->GetRotations();
+
+    scalingMatrix = glm::scale(
+        glm::mat4(1.0f),
+        glm::vec3(
+            scales[X], 
+            scales[Y], 
+            scales[Z]
+        )
+    );
+
+    rotationMatrix = glm::mat4(1.0f);
+    rotationMatrix = glm::rotate(
+        rotationMatrix,
+        glm::radians((float)rotations[X]),
+        glm::vec3(1.f, 0.f, 0.f)
+    );
+    rotationMatrix = glm::rotate(
+        rotationMatrix,
+        glm::radians((float)rotations[Y]),
+        glm::vec3(0.f, 1.f, 0.f)
+    );
+    rotationMatrix = glm::rotate(
+        rotationMatrix,
+        glm::radians((float)rotations[Z]),
+        glm::vec3(0.f, 0.f, 1.f)
+    );
+
+    translationMatrix = glm::translate(
+        glm::mat4(1.f),
+        glm::vec3(
+            translations[X],
+            translations[Y],
+            translations[Z]
+        )
+    );
+
+    translateOriginMatrix = glm::translate(
+        glm::mat4(1.f),
+        glm::vec3(
+            origins[X],
+            origins[Y],
+            origins[Z]
+        )
+    );
+}
+
+void AnnotationRenderer::_applyDataMgrCornerToDomain(
+    std::vector<double> &domainExtents,
+    const glm::vec4 dataMgrCorner,
+    const glm::mat4 scalingMatrix,
+    const glm::mat4 rotationMatrix,
+    const glm::mat4 translationMatrix,
+    const glm::mat4 translateOriginMatrix
+) const {
+    // transform our corner
+    glm::vec4 transformedCorner;
+    transformedCorner = glm::inverse(translateOriginMatrix) * dataMgrCorner;
+    transformedCorner = rotationMatrix * transformedCorner;
+    transformedCorner = translationMatrix * transformedCorner;
+    transformedCorner = scalingMatrix * transformedCorner;
+    transformedCorner = translateOriginMatrix * transformedCorner;
```

Comment:
Matrices can be concatenated. You are supposed to use them like this:

```
M = identity
M *= trans
M *= rot
M *= scale

Final vert = M * input vert 
```

Reply:
Interesting.  That's cleaner and I'll adopt this for my vectors and matrices moving on.